FastSimulation/ShowerDevelopment/interface/FastHFShowerLibrary.h

```diff
@@ -45,61 +46,32 @@ class FastHFShowerLibrary {
 
 public:
 
-  struct Hit {
-    Hit() {}
-    G4ThreeVector             position;
-    int                       depth;
-    double                    time;
-  };
-
   void                initRun();
   void       const    initHFShowerLibrary(const edm::EventSetup& );
   void                recoHFShowerLibrary(const FSimTrack &myTrack);
-  std::vector<Hit>    getHits(const G4ThreeVector & p, const G4ThreeVector & v,
-                              int parCode, double parEnergy, bool &ok, 
-                              double weight, bool onlyLong=false);
-  const std::map<CaloHitID,float>& getHitsMap() { return hitMap; };
-  bool                isItinFidVolume (G4ThreeVector&);
 
-protected:
+  std::vector<HFShowerLibrary::Hit> getHits(G4ThreeVector & p, G4ThreeVector & v,
+                                            int parCode, double parEnergy, bool &ok, 
+                                            double weight, bool onlyLong=false);
 
-  bool                rInside(double r);
-  void                getRecord(int, int);
-  void                loadEventInfo(TBranch *);
-  void                interpolate(int, double);
-  void                extrapolate(int, double);
-  void                storePhoton(int j);
-  std::vector<double> getDDDArray(const std::string&, const DDsvalues_type&,
-				  int&);
+  const std::map<CaloHitID,float>& getHitsMap() { return hitMap; };
 
-private:
+  bool                isItinFidVolume (G4ThreeVector&);
```

Comment:
@Vlandr57 

remove this function and replace it inside FastHFShowerLibrary::recoHFShowerLibrary
with 
if( !applyFidCut || applyFidCut(pos) )